clang/include/clang/Basic/BuiltinsAMDGPU.def

```diff
@@ -151,7 +151,7 @@ BUILTIN(__builtin_amdgcn_mqsad_u32_u8, "V4UiWUiUiV4Ui", "nc")
 //===----------------------------------------------------------------------===//
 
 TARGET_BUILTIN(__builtin_amdgcn_ballot_w32, "ZUib", "nc", "wavefrontsize32")
-TARGET_BUILTIN(__builtin_amdgcn_ballot_w64, "WUib", "nc", "wavefrontsize64")
+BUILTIN(__builtin_amdgcn_ballot_w64, "WUib", "nc")
```

Comment:
I am not sure whether this will cause the intrinsic being removed by backend if the required target feature is missing. I remember some pass is added for that.

Also removing the check increases chance of miss-use in HIP or OpenCL.

device libs bypass this requirement by adding `__attribute__((target("wavefrontsize64")))` to the callers. can you do the same?

Reply:
The difference is simply on the return value from the intrinsic. It's always legal to do type promotion, so on a system with a 32 wide wavefront it will just get promoted to a 64-bit value which will be correct.

The ROCm-Device-Libs can only do this because they have the `__oclc_wavefrontsize64` variable, which I don't want to copy.

Reply:
Just for reference, the main difference is whether or not we read `exec_hi`, here's some code https://godbolt.org/z/z64Yoo7ve. I believe @arsenm said that `exec_hi` is simply zero when read from wave32 mode? I could double check that.

The problem with using the attributes is that it puts the attribute into the function, which prevents it from being called in `wave32` correctly. This would require some compile-time switch like the ROCm libs, or perhaps a runtime check on the wave size.

Reply:
Did some basic checks and it works as expected when called from w32 code.

Reply:
The target attribute is just a giant footgun for AMDGPU. This one only works in the library with special handholding by the compiler